### PR TITLE
Expose diagnostics

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Adds `errors` and `warnings` to output of `transform()` function
- Adds `errors` and `warnings` to output of `convertToTSX()` function
- Adds `errors` and `warnings` to output of `parse()` function
- Replace `panic`s in the compiler

Will require a follow-up in core to surface these errors.

## Testing

Tests added for new behavior

## Docs

N/A

### CODE REVIEW

**Review by matthewp — APPROVED** [review]
Nice refactor.

**Comment by natemoo-re** [comment]
Just waiting on #542 before releasing a minor.

File: .changeset/curly-beers-raise.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+const { errors } = await transform(file, opts);
+for (const error of errors) {
+  console.error(error.text);
+}
+// or
+const { code, warnings } = await transform(file, opts);
+for (const warning of warnings) {
+  console.warn(warning.text);
+}
+```
```
Comment (Princesseuh) [inline .changeset/curly-beers-raise.md:19]: Not a big deal, but the changeset is outdated for the new return shape

Comment (natemoo-re) [reply .changeset/curly-beers-raise.md:19]: Great catch, thank you!

**Review by Princesseuh — APPROVED** [review]
Looks good to me, I haven't tested it in the AST context, but in the TSX output there's no apparent issues!

Very nice work, as always

**Pull request merged**